pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/render.go

```diff
@@ -178,15 +192,29 @@ func (r *components) renderComponent(ctx context.Context, acc v1alpha2.Applicati
 		traits = append(traits, &Trait{Object: *t, Definition: *traitDef})
 		traitDefs = append(traitDefs, *traitDef)
 	}
+	// we have a complete different approach on workload name for application generated appConfig
+	if !controlledByApp {
+		existingWorkload, err := r.getExistingWorkload(ctx, ac, c, w)
+		if err != nil {
+			return nil, err
+		}
 
-	existingWorkload, err := r.getExistingWorkload(ctx, ac, c, w)
-	if err != nil {
-		return nil, err
-	}
+		if err := SetWorkloadInstanceName(traitDefs, w, c, existingWorkload); err != nil {
+			return nil, err
+		}
+	} else {
+		revision, err := utils.ExtractRevision(acc.RevisionName)
+		if err != nil {
+			return nil, err
+		}
 
-	if err := SetWorkloadInstanceName(traitDefs, w, c, existingWorkload); err != nil {
-		return nil, err
+		pv := fieldpath.Pave(w.UnstructuredContent())
+		if err := pv.SetString(instanceNamePath, utils.ConstructRevisionName(acc.ComponentName,
+			int64(revision))); err != nil {
+			return nil, errors.Wrapf(err, errSetValueForField, instanceNamePath, c.GetName())
+		}
```

Comment:
do we have a test case for this difference

Reply:
not yet, will add in the next PR